Hallo.AspNetCore/HalJsonGenerator.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Hallo.AspNetCore
+{
+    public static class HalJsonGenerator
+    {
+        public static async Task HalHandler(HttpContext context, Object halObject)
+        {
+            var json = await GenerateHalJson(context, halObject);
+            if (json == null)
+            {
+                var serializerOptions = context.RequestServices.GetRequiredService<JsonSerializerOptions>();
+                context.Response.ContentType = "application/json";
+                await JsonSerializer.SerializeAsync(context.Response.Body, halObject, halObject.GetType(),
+                    serializerOptions);
+                await context.Response.Body.FlushAsync();
+            }
+            context.Response.ContentType = "application/hal+json";
+            await context.Response.WriteAsync(json);
+        }
+
+        public static async Task<string> GenerateHalJson(HttpContext context, Object halObject,
```

Comment:
This is the main refactoring, in that it has been moved from the outputformatter into a separate class ina separate package.